File: trunk/emsi/asp-net-app/WebEMSOF/component/db/Class_db.pas
unit Class_db;

interface

uses
  mysql.data.mysqlclient,
  system.configuration,
  system.data;

const
  ID = '$Id$';

type
  TClass_db = class
  private
    { Private Declarations }
  protected
  public
    connection: mysqlconnection;
      //
      // The connection member should be protected rather than public once all db operations in WebEMSOF are performed via
      // descendants of this class.
      //
    constructor Create;
    procedure Close;
    procedure Open;
  end;

implementation

constructor TClass_db.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
  connection := mysqlconnection.Create;
  connection.ConnectionString := configurationmanager.AppSettings['db_connection_string'];
end;

procedure TClass_db.Close;
begin
  connection.Close;
end;

procedure TClass_db.Open;
begin
  if connection.State <> connectionstate.OPEN then begin
    connection.Open;
  end;
end;

end.

File: asp-net-app/ConEdLink/component/db/Class_db.cs
using System.Data;
using MySql.Data.MySqlClient;

using System.Configuration;



using System;

namespace Class_db
{
    public class TClass_db
    {
        public MySqlConnection connection = null;
        // The connection member should be protected rather than public once all db operations in WebEMSOF are performed via
        // descendants of this class.
        //Constructor  Create()
        public TClass_db() : base()
        {
            // TODO: Add any constructor code here
            connection = new MySqlConnection();
            connection.ConnectionString = ConfigurationManager.AppSettings["db_connection_string"];
        }
        public void Close()
        {
            connection.Close();
        }

        public void Open()
        {
            if (connection.State != ConnectionState.Open)
            {
                connection.Open();
            }
        }

    } // end TClass_db

}

namespace Class_db.Units
{
    public class Class_db
    {
        public const string ID = "$Id: Class_db.pas 2343 2008-05-28 21:19:37Z kevinanlipscomb $";
    } // end Class_db

}

